### PR TITLE
Implementação da Cadastro após Convite na Pilot

### PR DESCRIPTION
**Pull request by miguelssrs**
## Contexto
Quando o cliente recebe um convite para usar a dashboard, ele clica no link enviado por email e é enviado para uma tela onde ele pode digitar a senha e criar sua nova conta de acesso. Esse PR atrás essa tela para o escopo Pilot.

## Checklist
- [ ] Criação de página de Cadastro após Convite
- [ ] Validação de e-mail desativada

## Issues linkadas
- [ ] Resolves https://github.com/pagarme/pilot/issues/1048

## Como testar?
- Vá para a branch `add/create-account-after-invite`
- Rode o comando `cd packages/pilot` e depois `yarn start`
- Crie um convite para algum e-mail de testes que você tiver em Configurações > Equipe > Adicionar Usuário.
- Vá para seu e-mail e clique no link que for enviado.
- Substitua o `dashboard.pagar.me` do link para `localhost:3000`. Exemplo: 
```
http://localhost:3000/#/account/signup/invite?token=daksjhgfsdajkdls&email=oieusouamiguel 20@gmail.com
```
- Lembre-se de testar com email com +, exemplo: `oieusouamiguel+2019@gmail.com`. É isso aê!

### CODE REVIEW

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: adba61777e8ab6d9804bd90c353827ae96b06257
  * lways use the standard `scope: action description`, sempre em inglês  
    - The **scope** shouldn't have spaces. If the word is composite, use dashes.  
    - The **action** should be **add**, **remove**, **refactor**, **move**,  **bump**, **update** or **release**  
    - The **description** should contain a quick description of the change

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: adba61777e8ab6d9804bd90c353827ae96b06257
  * lways use the standard `scope: action description`, sempre em inglês  
    - The **scope** shouldn't have spaces. If the word is composite, use dashes.  
    - The **action** should be **add**, **remove**, **refactor**, **move**,  **bump**, **update** or **release**  
    - The **description** should contain a quick description of the change

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: adba61777e8ab6d9804bd90c353827ae96b06257
  * lways use the standard `scope: action description`, sempre em inglês  
    - The **scope** shouldn't have spaces. If the word is composite, use dashes.  
    - The **action** should be **add**, **remove**, **refactor**, **move**,  **bump**, **update** or **release**  
    - The **description** should contain a quick description of the change

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: adba61777e8ab6d9804bd90c353827ae96b06257
  * lways use the standard `scope: action description`, sempre em inglês  
    - The **scope** shouldn't have spaces. If the word is composite, use dashes.  
    - The **action** should be **add**, **remove**, **refactor**, **move**,  **bump**, **update** or **release**  
    - The **description** should contain a quick description of the change

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 3869b7719a86087f8cf2a24a891e902fa706dd57
  * lways use the standard `scope: action description`, sempre em inglês  
    - The **scope** shouldn't have spaces. If the word is composite, use dashes.  
    - The **action** should be **add**, **remove**, **refactor**, **move**,  **bump**, **update** or **release**  
    - The **description** should contain a quick description of the change

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 3869b7719a86087f8cf2a24a891e902fa706dd57
  * lways use the standard `scope: action description`, sempre em inglês  
    - The **scope** shouldn't have spaces. If the word is composite, use dashes.  
    - The **action** should be **add**, **remove**, **refactor**, **move**,  **bump**, **update** or **release**  
    - The **description** should contain a quick description of the change

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 3869b7719a86087f8cf2a24a891e902fa706dd57
  * lways use the standard `scope: action description`, sempre em inglês  
    - The **scope** shouldn't have spaces. If the word is composite, use dashes.  
    - The **action** should be **add**, **remove**, **refactor**, **move**,  **bump**, **update** or **release**  
    - The **description** should contain a quick description of the change

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 3869b7719a86087f8cf2a24a891e902fa706dd57
  * lways use the standard `scope: action description`, sempre em inglês  
    - The **scope** shouldn't have spaces. If the word is composite, use dashes.  
    - The **action** should be **add**, **remove**, **refactor**, **move**,  **bump**, **update** or **release**  
    - The **description** should contain a quick description of the change

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by vagnervst** [comment]
@Deivis @rsmelo Fiz o rebase desta branch com a do PR https://github.com/pagarme/pilot/pull/1357 para usar as alterações que fiz no `PasswordInput`, assim consegui implementar a confirmação e força de senha.

File: packages/pilot/public/locales/pt/translations.json [hunk 1]
```diff
@@ -144,7 +146,11 @@
     "required": "Campo obrigatório",
     "sign_up_action": "Cadastrar",
     "subtitle": "na dashboard FormerKit",
-    "title": "crie sua conta"
+    "title": "crie sua conta",
+    "user": {
+      "confirmation": "Seu usuário foi criado com sucesso! Retorne à página de login para começar a utilizá-lo.",
+      "confirmation_emphasis": "Pronto!"
```
Comment (uselessdev) [inline packages/pilot/public/locales/pt/translations.json]: Porque esse é capitalizado e outro não? 🤔

File: packages/pilot/src/containers/Account/SignUp/User/Form.js [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        </div>
+      </div>
+    </Form>
+  )
+}
+
+SignUpInvite.defaultProps = {
+  errors: null,
+  loading: false,
+  t: t => t,
```
Comment (uselessdev) [inline packages/pilot/src/containers/Account/SignUp/User/Form.js]: `t` precisava ser uma prop obrigatória, porque agora não é mais?

File: packages/pilot/src/pages/Account/SignUp/Confirmation.js [hunk 3]
```diff
@@ -23,11 +23,23 @@ class SignUpConfirmationPage extends PureComponent {
...
   render () {
-    const { t } = this.props
+    const {
+      model,
+      t,
+    } = this.props
+
+    const translationPath = model === 'company'
+      ? 'sign_up.company'
+      : 'sign_up.user'
```
Comment (uselessdev) [inline packages/pilot/src/pages/Account/SignUp/Confirmation.js]:
```
const translationPath = ifElse(
  equals('company'),
  always('sign_up.company'),
  always('sign_up.user')
)
```

ramda all the things

File: packages/pilot/src/pages/Account/index.js [hunk 4]
```diff
@@ -81,15 +82,23 @@ const AccountArea = ({ history: { location }, t }) => {
           />
           <Route
             path="/account/signup/confirmation"
-            component={SignUpConfirmation}
+            component={() => <SignUpConfirmation model="company" />}
```
Comment (uselessdev) [inline packages/pilot/src/pages/Account/index.js]: Aqui não seria melhor usar `render={props => <Component />}` no lugar de `component`?

File: packages/pilot/src/containers/Account/PasswordRecovery/Request/Confirmation.js [hunk 5]
```diff
@@ -1,45 +1,25 @@
 import React from 'react'
```
Comment (rsmelo) [inline packages/pilot/src/containers/Account/PasswordRecovery/Request/Confirmation.js]: será que aqui seria interessante renomear o arquivo para `RecoveryConfirmation.js`  pra ficar consistente com o componente?

File: packages/pilot/public/locales/pt/translations.json (hunk 1 above)
Comment (vagnervst) [reply packages/pilot/public/locales/pt/translations.json]: Boa, capitalizei o outro para manter o padrão

File: packages/pilot/src/containers/Account/SignUp/User/Form.js (hunk 2 above)
Comment (vagnervst) [reply packages/pilot/src/containers/Account/SignUp/User/Form.js]: Não sei o contexto dessa alteração, mudei para required novamente 🤔

File: packages/pilot/src/pages/Account/index.js (hunk 4 above)
Comment (vagnervst) [reply packages/pilot/src/pages/Account/index.js]: Como é uma página que é acessada ocasionalmente, não sei se vale a preocupação uólace 🤔 mas se achar que faz sentido posso mudar sem problemas

File: packages/pilot/src/pages/Account/SignUp/Confirmation.js (hunk 3 above)
Comment (vagnervst) [reply packages/pilot/src/pages/Account/SignUp/Confirmation.js]: but it got a lot more complicado D:

File: packages/pilot/src/pages/Account/index.js (hunk 4 above)
Comment (rsmelo) [reply packages/pilot/src/pages/Account/index.js]: concordo que faz mais sentido usar render sim

**Comment by rsmelo** [comment]
após preencher a senha e se cadastrar acontece o seguinte erro no console
`Warning: Failed prop type: The prop `labels.confirmationContent` is marked as required in `Confirmation`, but its value is `undefined`.`

File: packages/pilot/src/pages/Account/SignUp/Confirmation.js (hunk 3 above)
Comment (uselessdev) [reply packages/pilot/src/pages/Account/SignUp/Confirmation.js]: I think it's simpler than a ternary, but, its only a suggestion anyway. 👀

File: packages/pilot/src/pages/Account/index.js (hunk 4 above)
Comment (uselessdev) [reply packages/pilot/src/pages/Account/index.js]: Alias, tem mais abaixo o mesmo caso 😊

Comment (vagnervst) [reply packages/pilot/src/pages/Account/index.js]: Feito!

**Comment by vagnervst** [comment]
@rsmelo boa! Resolvido

File: packages/pilot/src/pages/Account/SignUp/User/index.js [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+      setErrors({
+        password: weakPassword,
+        passwordConfirm: weakPassword,
+      })
+
+      return
+    }
+
+    if (isEmpty(errors) && !validationErrors) {
+      setErrors({})
```
Comment (rsmelo) [inline packages/pilot/src/pages/Account/SignUp/User/index.js]: aqui precisa chamar `setErrors({})`? pois se ele já é empty no if da linha 104 ele ja vai ser `{}`

File: packages/pilot/src/pages/Account/SignUp/User/index.js [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+  useEffect(() => {
+    if (!token) {
+      history.replace('/account/login')
+    }
+  }, [history, token])
+
+  const validateInputs = ({ password = '', passwordConfirm = '' }) => {
+    const validations = validate(password)
+
+    const passwordMatch = password !== passwordConfirm
```
[the remark this answers is not in the document]
Comment (vagnervst) [reply packages/pilot/src/pages/Account/SignUp/User/index.js]: Feito!

File: packages/pilot/src/pages/Account/SignUp/User/index.js (hunk 6 above)
Comment (vagnervst) [reply packages/pilot/src/pages/Account/SignUp/User/index.js]: Boa, não é necessário mesmo. Removido!

**Comment by gmbarroso** [comment]
@vagnervst A branch está desatualizada. Só arrumar isso.

**Comment by vagnervst** [comment]
@Deivis Show, adicionei o Alert e refatorei para utilizar RX

**Comment by rsmelo** [comment]
@vagnervst o outro pr ja foi mergeado, pode atualizar esse =)

File: packages/pilot/src/pages/Account/SignUp/Company/Confirmation.js [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+import { translate } from 'react-i18next'
+import { compose } from 'ramda'
+import { CompanySignupConfirmation } from '../../../../containers/Account/SignUp/Company'
+
+const enhanced = compose(
+  translate(),
+  withRouter
+)
+
+class CompanySignupConfirmationPage extends PureComponent {
```
Comment (Deivis) [inline packages/pilot/src/pages/Account/SignUp/Company/Confirmation.js]: Este componente poeria ser mais simples e usar hooks 🤔

Comment (vagnervst) [reply packages/pilot/src/pages/Account/SignUp/Company/Confirmation.js]: Feito!

File: packages/pilot/src/pages/Account/SignUp/Confirmation.js [hunk 9]
```diff
@@ -3,7 +3,7 @@ import PropTypes from 'prop-types'
 import { withRouter } from 'react-router-dom'
 import { translate } from 'react-i18next'
 import { compose } from 'ramda'
-import { SignUpConfirmation } from '../../../containers/Account/SignUp'
+import Confirmation from '../../../containers/Account/Confirmation'
```
Comment (rsmelo) [inline packages/pilot/src/pages/Account/SignUp/Confirmation.js]: esse componente poderia ser um function componente

File: packages/pilot/src/pages/Account/actions/reducer.js [hunk 10]
```diff
@@ -85,6 +90,38 @@ export default function loginReducer (state = initialState, action) {
       )
     }
 
+    case USER_SIGNUP_REQUEST: {
```
Comment (rsmelo) [inline packages/pilot/src/pages/Account/actions/reducer.js]: por que teve necessidade de usar o reducer e épico pra gerenciar se a request de invite deu certo, apenas usar o state da pagina não seria suficiente?

**Comment by Deivis** [comment]
Também seria interessante fazer o bump do `react-vanilla-form` para evitar os seguintes `logs`

File: packages/pilot/src/pages/Account/actions/reducer.js (hunk 10 above)
Comment (vagnervst) [reply packages/pilot/src/pages/Account/actions/reducer.js]: Feito! Deixei apenas a state do component

**Comment by vagnervst** [comment]
@Deivis @rsmelo O problema com a prop type do `score` eu resolvo nesse PR: https://github.com/pagarme/p4g4rm3/pull/21

@Deivis Não consegui reproduzir esses logs aqui 🤔

**Comment by Deivis** [comment]
> @Deivis @rsmelo O problema com a prop type do `score` eu resolvo nesse PR: [pagarme/p4g4rm3#21](https://github.com/pagarme/p4g4rm3/pull/21)
> 
> @Deivis Não consegui reproduzir esses logs aqui 🤔

Podes ignorar os logs, era minha versão do `react-vanilla-form` que estava com o log.
Mas a atualização do `react-vanilla-form` ainda acho que pode ser feita 😄

File: packages/pilot/src/pages/Account/actions/reducer.js [hunk 11]
```diff
@@ -22,6 +22,7 @@ const getBalance = applySpec({
 const initialState = {
   loading: false,
   sessionId: null,
+  userSignup: false,
```
Comment (rsmelo) [inline packages/pilot/src/pages/Account/actions/reducer.js]: precisa desse false aqui ainda? essa prop não acabou morrendo sem a necessidade do reducer?

File: packages/pilot/stories/pages/Landing/index.js [hunk 12]
```diff
@@ -188,14 +191,30 @@ storiesOf('Pages|Signup', module)
       t={t}
     />
   ))
-  .add('Signup E-mail Invalid', () => (
+  .add('User Signup', () => (
+    <Account
+      // eslint-disable-next-line
```
Comment (rsmelo) [inline packages/pilot/stories/pages/Landing/index.js]: por que esse disable aqui?

**Comment by rsmelo** [comment]
@vagnervst se tento acessar a url sem passar um email como parametro não da erro porem o campo de email está vazio e não me permite digitar. 
#### como reproduzir
- subir a pilot  `yarn start`
- acessar a url `http://localhost:3000/#/account/signup/invite?token=aaaaaa`

File: packages/pilot/src/pages/Account/actions/reducer.js (hunk 11 above)
Comment (vagnervst) [reply packages/pilot/src/pages/Account/actions/reducer.js]: Resolvido

**Comment by vagnervst** [comment]
@rsmelo Corrigido o ponto do e-mail, agora faço a página redirecionar para o login caso o e-mail não exista

**Pull request merged**